frontend/lib/resolution/resolution-queries.cpp

```diff
@@ -109,7 +105,44 @@ scopeResolveModuleStmt(Context* context, ID id) {
   return QUERY_END(result);
 }
 
+static void updateTypeForModuleLevelSplitInit(Context* context, ID id,
+                                   ResolvedExpression& lhs,
+                                   const ResolvedExpression& rhs,
+                                   std::set<ID>& alreadyUpdated) {
+  const QualifiedType lhsType = lhs.type();
+  const QualifiedType rhsType = rhs.type();
+
+  // check to see if it is generic/unknown
+  // (otherwise we do not need to infer anything)
+  if (!(lhsType.isUnknownKindOrType() ||
+        getTypeGenericity(context, lhsType.type()) == Type::GENERIC))
+    return;
 
+  const Param* p = rhsType.param();
+  if (lhsType.kind() != QualifiedType::PARAM) {
+    p = nullptr;
+  }
+  const auto useType = QualifiedType(lhsType.kind(), rhsType.type(), p);
+
+  // set the type for the 1st split init only
+  // a later traversal will check the type of subsequent split inits
+  // (in the other branch of a conditional, say)
+  auto pair = alreadyUpdated.insert(id);
+  if (pair.second) {
+    // insertion took place, so update the type
+    lhs.setType(useType);
+  } else {
+    // insertion did not take place, so check that the type matches exactly,
+    // and issue an error if not.
+    // (we cannot unify the types for split init without causing resolution
+    //  to either go out of order or to produce results that change within
+    //  a function even when there are no errors).
+
+    if (lhsType != useType) {
+      context->error(id, "split-init type does not match");
+    }
```

Comment:
Is this not the same error as `SplitInitMismatchedConditionalTypes`?

Reply:
I'm actually not sure, I had forgotten about this spot. I don't know how to make this error happen. This doesn't get hit for a simple if/else with branches initializing a variable to different types:
```
var flag = true;
var foo;
if (flag) {
  foo = 5;
} else {
  foo = 5.5;
}
```

Or for this case (which actually doesn't emit any error in `testInteractive` 😕 ):
```
var foo;
foo = 5;
foo = "string";
```

Reply:
Tried some more to make it come up (including with the above examples in a `main` in case this is just a problem for module-level code) and couldn't. There's no testing for this particular error based on grepping for the message, nor was there when it was first introduced. I _think_ it isn't possible to hit, because we could only reach that point if the declaration in question is generic or unknown, which it wouldn't be if we've already split-init'd it once.

So, I would like to just remove this error and the checking for already-inferred variables entirely. Does that seem reasonable? FWIW, doing so doesn't cause any test failures for me.

Reply:
I'm all for deleting code, especially if it's dead. If you can do it without causing errors, I'd say go ahead!